streaming/src/main/scala/org/apache/spark/status/api/v1/streaming/StreamingApiRootResource.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.status.api.v1.streaming
+
+import javax.ws.rs.{Path, PathParam}
+
+import org.apache.spark.status.api.v1.{NotFoundException, UIRootFromServletContext}
+import org.apache.spark.streaming.ui.StreamingJobProgressListener
+import org.apache.spark.ui.SparkUI
+
+@Path("/v1")
+private[v1] class StreamingApiRootResource extends UIRootFromServletContext {
+
+  import org.apache.spark.status.api.v1.streaming.StreamingApiRootResource._
+
+  @Path("applications/{appId}/streaming/statistics")
```

Comment:
So, you'll also need variants that take `{appId}/{attemptId}` arguments, otherwise this won't work in certain deploy modes (like yarn-cluster). To avoid too much copy / paste, it might be better to add a level of indirection (names are just suggestions):

```
class ApiStreamingApp {

  @Path("applications/{appId}/streaming")
  def getStreamingRoot(...): ApiStreamingRootResource = {
    uiRoot.withSparkUI(appId, None) { new ApiStreamingRootResource(_) }
  }

  @Path("applications/{appId}/{attemptId}/streaming")
  def getStreamingRoot(...): ApiStreamingRootResource = {
    uiRoot.withSparkUI(appId, attemptId) { new ApiStreamingRootResource(_) }
  }
```

Then you can reuse `ApiStreamingRootResource` for both apps that have attemptIds and those that do not.